chia/full_node/coin_store.py

```diff
@@ -429,8 +461,12 @@ async def _set_spent(self, coin_names: List[bytes32], index: uint32):
                 self.coin_record_cache.put(
                     r.name, CoinRecord(r.coin, r.confirmed_block_index, index, True, r.coinbase, r.timestamp)
                 )
-            updates.append((index, coin_name.hex()))
-
-        await self.coin_record_db.executemany(
-            "UPDATE OR FAIL coin_record SET spent=1,spent_index=? WHERE coin_name=?", updates
-        )
+            updates.append((coin_name.hex()))
+
+        if len(updates) > 0:
+            cursor = await self.coin_record_db.execute(
+                f'UPDATE coin_record '
+                f'   SET spent=1, spent_index=? '
+                f' WHERE coin_name in ({"?," * (len(coin_names) - 1)}?)',
+                (index,) + tuple(updates),
```

Comment:
the default build of sqlite on windows has a pretty low limit of the number of arguments per query. a large block may have more more removals than allowed by that limit, so we can't do this in a single statement

Reply:
This pattern is used in a couple other places, same issue.  In the [sqlite documentation](https://www.sqlite.org/limits.html) I am reading `To prevent excessive memory allocations, the maximum value of a host parameter number is SQLITE_MAX_VARIABLE_NUMBER, which defaults to 999 for SQLite versions prior to 3.32.0 (2020-05-22) or 32766 for SQLite versions after 3.32.0.`  32766 is actually pretty big.  Do you think that limit will be passed anytime soon also? Then other areas of the code need refactoring also, some kind of chunking.

Reply:
Iirc, one of the default python builds we support for windows has the older 999 limit. A single block can contain upwards of 2000 additions and removals